exercises/gagik_vardanyan/ex02_oo_basics/src/main.cpp

```diff
@@ -14,22 +14,17 @@
 ///the main program
 int main(int argc, char ** argv)
 {
-    std::vector<Shape> shapes;
+    std::vector<Shape*> shapes;
 
     Rectangle * rect = new Rectangle(10,50);
     Triangle * tri = new Triangle(10,5);
     Square * sq = new Square(7);
```

Comment:
You could have made these shapes constant. Your sorting does not (and should not) modify the Shapes in the vector, so I would suggest enforcing that by storing a `const Shape*` in the `vector`